includes/export/class-wc-product-csv-exporter.php

```diff
@@ -176,9 +176,11 @@ public function prepare_data_to_export() {
 		$this->row_data    = array();
 		$variable_products = array();
 
+		$fetch_variations = apply_filters( "woocommerce_product_export_{$this->export_type}_fetch_variations", ! empty( $args['category'] ), $args );
+
 		foreach ( $products->products as $product ) {
-			// Check if the category is set, this means we need to fetch variations seperately as they are not tied to a category.
-			if ( ! empty( $args['category'] ) && $product->is_type( 'variable' ) ) {
+			// Check if the category is set, this means we need to fetch variations separately as they are not tied to a category.
+			if ( $product->is_type( 'variable' ) && $fetch_variations ) {
```

Comment:
Doing the change I suggested you can remove the comment from this check, and also need to update to:

```php
if ( $is_variable ) {
```